Code/GraphMol/ChemTransforms/MolFragmenter.cpp

```diff
@@ -739,17 +739,34 @@ struct ZipBond {
                       "molzip: begin atom and specified dummy atom connection "
                       "are not bonded.")
       auto bond_type_a = bnd->getBondType();
+      auto bond_dir_a = bnd->getBondDir();
       bnd = newmol.getBondBetweenAtoms(b->getIdx(), b_dummy->getIdx());
       CHECK_INVARIANT(bnd != nullptr,
                       "molzip: end atom and specified dummy connection atom "
                       "are not bonded.")
       auto bond_type_b = bnd->getBondType();
+      auto bond_dir_b = bnd->getBondDir();
+    
+      unsigned int bnd_idx = 0;
       if (bond_type_a != Bond::BondType::SINGLE) {
-        newmol.addBond(a, b, bond_type_a);
+        bnd_idx = newmol.addBond(a, b, bond_type_a);
       } else if (bond_type_b != Bond::BondType::SINGLE) {
-        newmol.addBond(a, b, bond_type_b);
+        bnd_idx = newmol.addBond(a, b, bond_type_b);
       } else {
-        newmol.addBond(a, b, Bond::BondType::SINGLE);
+        bnd_idx = newmol.addBond(a, b, Bond::BondType::SINGLE);
+      }
+        
+      if(bond_dir_a != Bond::NONE || bond_dir_b != Bond::NONE) {
+          auto bond_dir = Bond::NONE;
+          if (bond_dir_a != Bond::NONE) {
+              bond_dir = bond_dir_a;
+          } else if (bond_dir_b != Bond::NONE) {
+              bond_dir = bond_dir_b;
+          } else {
+              BOOST_LOG(rdWarningLog) << "imcompatible bond directions when zipping, choosing bond direction from first fragment" << std::endl;
+              bond_dir = bond_dir_a;
+          }
```

Comment:
I don't think you can ever get to this clause of the if statement, on line 759 you have checked that at least one of the two bonds does not have direction BOND_NONE
```suggestion
          }
```

Reply:
Also, it seems like you're going to need to worry about the begin and end atoms of the bond if you take the direction from bond_b.
Supposing you zip together `CC[*]` and `N[*]`, as in your test, but the wedge is on the `N-[*]` bond.
In the ouptput molecule I think the new `C-N` bond is going to start at the `C` atom instead of the 'N`, so the wedging is going to be applied to the wrong atom. So I think here you need to check where the begin/end atoms of the bond are and adjust them accordingly

Reply:
All of these were good issues to catch, I've added better logic, hopefully it is correct :)